### PR TITLE
feat(MessageAttachment): add support for ephemeral property

### PR DESCRIPTION
**Pull request by iShibi**
**Please describe the changes this PR makes and why it should be merged:**

- discord/discord-api-docs#3833

**Status and versioning classification:**
- I know how to update typings and have done so
- Code changes have been tested against the Discord API

### CODE REVIEW

**Comment by iShibi** [comment]
This is working as expected:
```js
client.on('interactionCreate', async interaction => {
    if (!interaction.isCommand()) return;
    if (interaction.commandName === 'test') {
        const attachment = new MessageAttachment('./pic.jpg', 'pic.jpg');
        await interaction.reply({ ephemeral: true, files: [attachment] }); // ✔️ works
    }
});
```
but it seems like you can't send attachments in a deferred ephemeral interaction:
```js
client.on('interactionCreate', async interaction => {
    if (!interaction.isCommand()) return;
    if (interaction.commandName === 'test') {
        await interaction.deferReply({ ephemeral: true })
        const attachment = new MessageAttachment('./pic.jpg', 'pic.jpg');
        await interaction.editReply({ files: [attachment] }); // ✖️ doesn't work
    }
});
```

**Comment by conman0552** [comment]
Hopefully this gets released soon. I know I'm in need of this feature for some specific things with embeds, and this would be awsome!

File: src/structures/MessageAttachment.js [hunk 1]
```diff
@@ -103,6 +103,12 @@ class MessageAttachment {
      * @type {?string}
      */
     this.contentType = data.content_type ?? null;
+
+    /**
+     * Whether this attachment is ephemeral
+     * @type {?boolean}
+     */
+    this.ephemeral = data.ephemeral ?? null;
```
Comment (ImRodry) [inline src/structures/MessageAttachment.js:111]: I believe `data.ephemeral` is only sent when true, so shouldn't this default to false instead?

**Review by SpaceEEC — APPROVED** [review]

**Review by Mysterious-Dev — APPROVED** [review]

**Comment by iShibi** [comment]
> ⚠️ **Note: The following issue has been fixed, don't waste your time reading this comment** ⌛
> 
> This is working as expected:
> 
> ```js
> client.on('interactionCreate', async interaction => {
>     if (!interaction.isCommand()) return;
>     if (interaction.commandName === 'test') {
>         const attachment = new MessageAttachment('./pic.jpg', 'pic.jpg');
>         await interaction.reply({ ephemeral: true, files: [attachment] }); // ✔️ works
>     }
> });
> ```
> 
> but it seems like you can't send attachments in a deferred ephemeral interaction:
> 
> ```js
> client.on('interactionCreate', async interaction => {
>     if (!interaction.isCommand()) return;
>     if (interaction.commandName === 'test') {
>         await interaction.deferReply({ ephemeral: true })
>         const attachment = new MessageAttachment('./pic.jpg', 'pic.jpg');
>         await interaction.editReply({ files: [attachment] }); // ✖️ doesn't work
>     }
> });
> ```

Checked again and found this has been fixed ✨
My guess?? https://github.com/discord/discord-api-docs/issues/3842#issuecomment-923249437 🤷

File: src/structures/MessageAttachment.js (hunk 1 above)
Comment (iShibi) [reply src/structures/MessageAttachment.js:111]: yeah, I guess so. What do others think should be done here? 👀

**Review by iCrawl — APPROVED** [review]

File: src/structures/MessageAttachment.js (hunk 1 above)
Comment (vladfrangu) [reply src/structures/MessageAttachment.js:111]: Personally I agree. @discordjs/the-big-3?

Comment (iCrawl) [reply src/structures/MessageAttachment.js:111]: Yes.

**Review by kyranet — APPROVED** [review]

**Pull request merged**